astropy/config/paths.py

```diff
@@ -167,48 +167,40 @@ def _get_dir_path(cls, rootname: str) -> Path:
                 path.mkdir(exist_ok=True)
             return path.resolve()
 
-        if (
-            (dir_ := os.getenv(cls._directory_env_var)) is not None
-            and (xch := Path(dir_)).exists()
-            and not (xchpth := xch / rootname).is_symlink()
-        ):
-            if xchpth.exists():
-                return xchpth.resolve()
+        env_dir = cls._xdg_env_dir
+        if (dir_ := os.getenv(env_dir)) is None:
+            return cls._find_or_create_root_dir(rootname)
 
-            # symlink will be set to this if the directory is created
-            linkto = xchpth
-        else:
-            linkto = None
+        if not (xch := Path(dir_)).is_dir():
+            warn(
+                f"{env_dir} is set to {dir_!r}, but no such directory was found. "
+                "So, this environment variable is ignored.",
+                category=UserWarning,
+                stacklevel=3,
+            )
+            return cls._find_or_create_root_dir(rootname)
 
-        return cls._find_or_create_root_dir(linkto, rootname)
+        xchpth = xch / rootname
+        xchpth.mkdir(exist_ok=True)
+        return xchpth.resolve()
 
     @classmethod
-    def _find_or_create_root_dir(
-        cls,
-        linkto: Path | None,
-        pkgname: str = "astropy",
-    ) -> Path:
-        innerdir = Path.home() / f".{pkgname}"
+    def _find_or_create_root_dir(cls, rootname: str = "astropy") -> Path:
+        innerdir = Path.home() / f".{rootname}"
         maindir = innerdir / cls._directory_type
 
         if maindir.is_file():
             raise OSError(
-                f"Intended {pkgname} {cls._directory_type} directory {maindir} is actually a file."
+                f"Intended {rootname} {cls._directory_type} directory {maindir} is actually a file."
             )
-        if not maindir.is_dir():
-            # first create .astropy dir if needed
-            if innerdir.is_file():
-                raise OSError(
-                    f"Intended {pkgname} {cls._directory_type} directory {maindir} is actually a file."
-                )
-            maindir.mkdir(parents=True, exist_ok=True)
-            if (
-                not sys.platform.startswith("win")
-                and linkto is not None
-                and not linkto.exists()
-            ):
-                linkto.symlink_to(maindir)
```

Comment:
Yeah, I don't think we can delete this. Seems like it is intentional in original design. Changing the design is out of scope here. From the docstring.

```rst
    This directory is typically ``$HOME/.astropy/config``, but if the
    XDG_CONFIG_HOME environment variable is set and the
    ``$XDG_CONFIG_HOME/astropy`` directory exists, it will be that directory.
    If neither exists, the former will be created and symlinked to the latter.
```

Reply:
Feel free to drop the last commit then

Reply:
Do you want me to take over? I am not sure how to interpret your reply. 😅

Reply:
I'm not near a keyboard right now. I'm giving you explicit permission to edit the branch while I'm away. I can do it myself in about 30min though.